extensions/amp-story/0.1/media-pool.js

```diff
@@ -92,6 +212,48 @@ function ampMediaElementFor(el) {
 }
 
 
+/**
+ * @param {!HTMLMediaElement} mediaEl The element whose task queue should be
+ *     executed.
+ */
+function executeNextMediaElementTask(mediaEl) {
+  const queue = mediaEl[ELEMENT_TASK_QUEUE_PROPERTY_NAME];
+
+  if (queue.length === 0) {
+    return;
+  }
+
+  const taskName = queue[0];
+  const task = ELEMENT_TASKS[taskName] || NOOP_ELEMENT_TASK;
+
+  task(mediaEl)
+      .catch(reason => dev().error('AMP-STORY', reason))
+      .then(() => {
+        queue.shift();
+        executeNextMediaElementTask(mediaEl);
+      });
+}
+
+
+/**
+ * @param {!HTMLMediaElement} mediaEl The element for which the specified task
+ *     should be executed.
+ * @param {!ElementTaskName} taskName The name of the task to execute.
+ */
+function enqueueMediaElementTask(mediaEl, taskName) {
+  if (!mediaEl[ELEMENT_TASK_QUEUE_PROPERTY_NAME]) {
+    mediaEl[ELEMENT_TASK_QUEUE_PROPERTY_NAME] = [];
+  }
+
+  const queue = mediaEl[ELEMENT_TASK_QUEUE_PROPERTY_NAME];
+  queue.push(taskName);
+
+  if (queue.length === 1) {
```

Comment:
Nit: might be worth implementing a ``isQueueRunning()`` or ``shouldExecuteQueue`` or something helper, that would do just that

Reply:
made a constant